python/private/stage2_bootstrap_template.py

```diff
@@ -365,6 +369,21 @@ def main():
     print_verbose("initial environ:", mapping=os.environ)
     print_verbose("initial sys.path:", values=sys.path)
 
+    site_packages = os.path.join(sys.prefix, VENV_SITE_PACKAGES)
+    if site_packages not in sys.path and os.path.exists(site_packages):
+        # NOTE: if this happens, it likely means we're running with a different
+        # Python version than was built with. Things may or may not work.
```

Comment:
It is very likely that all of the non-pure python packages will stop working. I think we should error and tell the user a more helpful message than what is normally present.

Reply:
I tend to agree, however, it's at odds with the runtime environment tool chain behavior. (That's the only reason I put this here)

How about only do this logic when supports_build_time_venv is false (essentially just the runtime environment tool chain)?  That case is problematic/hacky in other ways anyways already; pretty much everything around it is just doing what it can and trusting the user knows what they're doing.

Reply:
Updated this logic to only run if build-time venv isn't supported by the toolchain.

Reply:
Thank you,